articles/142_idl.md

```diff
@@ -144,3 +144,95 @@ For now only a one-dimensional, fixed-size array is supported though.
 <div class="alert alert-warning" markdown="1">
   <b>TODO:</b> All of this too
 </div>
+
+### Constraint Checking
+
+Generated code for messages is only guaranteed to enforce constraints when a message is published.
+For example, there might not be an error when a field constrained to be a fixed size array is assigned an array with the wrong size.
+This could be inconsistent across client library implementations depending on language features and performance cost.
+In the future any client library may add additional constraint checking when a field is set or modified.
+Users of generated message code should assume constraints could be enforced at any time to be compatible with such a change.
+
+### Type Mapping
+
+The next table defines how IDL types are mapped to the following programming languages:
+
+* C11 (or higher)
+* C++11 (or higher)
+* Python 3 (or higher)
+
+| IDL type    | C type           | C++ type                       | Python type          |
+| --------    | -----------      | ------------------------------ | -------------------- |
+| float       | float            | float                          | float                |
+| double      | double           | double                         | float                |
+| long double | long double      | long double[<sup>2</sup>](#ld) | float                |
+| char        | signed char      | char                           | str with length 1    |
```

Comment:
IDL `7.2.6.2.1` says `A char is an 8-bit quantity with a numerical value between 0 and 255 (decimal)`. Changing to `unsigned char` in c and c++